code/modules/projectiles/guns/energy/laser_musket.dm

```diff
@@ -0,0 +1,32 @@
+/obj/item/gun/energy/laser/musket
+	name = "laser musket"
+	desc = "A hand-crafted laser weapon, it has a hand crank on the side to charge it up."
+	icon_state = "musket"
+	inhand_icon_state = "musket"
+	worn_icon_state = "las_musket"
+	ammo_type = list(/obj/item/ammo_casing/energy/laser/musket)
+	slot_flags = ITEM_SLOT_BACK
+	obj_flags = UNIQUE_RENAME
+	var/is_charging = FALSE
+	can_bayonet = TRUE
+	knife_x_offset = 22
+	knife_y_offset = 11
+
+/obj/item/gun/energy/laser/musket/attack_self(mob/living/user as mob)
+	var/obj/item/stock_parts/cell/charging_cell = get_cell()
+	if(charging_cell.charge < charging_cell.maxcharge)
+		if(is_charging == FALSE)
+			is_charging = TRUE
+			playsound(src, 'sound/weapons/laser_crank.ogg', 40)
+		balloon_alert(user, "charging...")
+		if(do_after(user, 2 SECONDS, src, interaction_key = DOAFTER_SOURCE_CHARGE_MUSKET))
+			charging_cell.give(500)
+			update_appearance()
+			is_charging = FALSE
+			balloon_alert(user, "charged")
+	else
+		balloon_alert(user, "already charged!")
```

Comment:
You should try and reformat this to use early returns. Message me on discord if you need further explanation on how that would work.

Reply:
Done this to the best of my ability, thank you!!!